src/main/c/jpostal_Dedupe.c

```
@@ -141,3 +141,107 @@ JNIEXPORT void JNICALL Java_com_mapzen_jpostal_DuplicateOptions_00024Builder_set
 
     (*env)->SetObjectField(env, builder, fid, NULL);
 }
+
+typedef libpostal_fuzzy_duplicate_status_t (*fuzzy_duplicate_function)(size_t, char **, double *, size_t, char **, double *, libpostal_fuzzy_duplicate_options_t);
+
+JNIEXPORT jdouble JNICALL Java_com_mapzen_jpostal_Dedupe_isDuplicateFuzzy(
+    JNIEnv *env, jobject thisObj,
+    jobjectArray jTokens1, jdoubleArray jScores1,
+    jobjectArray jTokens2, jdoubleArray jScores2,
+    jobject jOptions, fuzzy_duplicate_function func
+) {
+    jboolean is_copy = JNI_FALSE;
+
+    // Get arguments
+    int num_tokens1 = (*env)->GetArrayLength(env, jTokens1);
+    int num_tokens2 = (*env)->GetArrayLength(env, jTokens2);
+
+    double *scores1 = (*env)->GetDoubleArrayElements(env, jScores1, &is_copy);
+    double *scores2 = (*env)->GetDoubleArrayElements(env, jScores2, &is_copy);
+
+    const char *tokens1[num_tokens1];
+    const char *tokens2[num_tokens2];
+
+    for (int i = 0; i < num_tokens1; i++) {
+        jstring jToken = (jstring) (*env)->GetObjectArrayElement(env, jTokens1, i);
+        const char *token = (*env)->GetStringUTFChars(env, jToken, &is_copy);
+        tokens1[i] = strdup(token);
+        (*env)->ReleaseStringUTFChars(env, jToken, token);
+    }
+
+    for (int i = 0; i < num_tokens2; i++) {
+        jstring jToken = (jstring) (*env)->GetObjectArrayElement(env, jTokens2, i);
+        const char *token = (*env)->GetStringUTFChars(env, jToken, &is_copy);
+        tokens2[i] = strdup(token);
+        (*env)->ReleaseStringUTFChars(env, jToken, token);
+    }
+
+    // Build Options Argument
+    libpostal_fuzzy_duplicate_options_t options = libpostal_get_default_fuzzy_duplicate_options();
+
+    jfieldID fid;
+    jclass optionsCls = (*env)->GetObjectClass(env, jOptions);
+
+    fid = (*env)->GetFieldID(env, optionsCls, "languages", "[Ljava/lang/String;");
+    if (fid == 0) {
+        return 0.0;
```

I'm not sure what this condition is checking, but I would think it's better to throw an exception or ignore instead of returning what could be interpreted as a valid score of 0.

---

Good call, updated the code to throw an exception